core/domain/exp_services.py

```diff
@@ -1622,6 +1623,43 @@ def get_next_page_of_all_non_private_commits(
     ) for entry in results], new_urlsafe_start_cursor, more)
 
 
+def get_image_filenames_from_exploration(exploration):
+    """ Get the image filenames from the exploration.
+
+    Args:
+        exploration: object. The exploration itself.
+
+    Returns:
+       list. The list containing the name of the image files in the exploration
+    """
+    filenames = []
+    for state in exploration.states.itervalues():
+        if state.interaction.id == 'ImageClickInput':
+            if isinstance(state.interaction.customization_args['imageAndRegions']['value']['imagePath'], dict): # pylint: disable=line-too-long
+                filenames.append(state.interaction.customization_args['imageAndRegions']['value']['imagePath']['name']) # pylint: disable=line-too-long
+            else:
+                filenames.append(state.interaction.customization_args['imageAndRegions']['value']['imagePath']) # pylint: disable=line-too-long
+
+    html_list = exploration.get_all_html_content_strings()
+    rte_components_in_exp = []
+    for html_string in html_list:
+        rte_components_in_exp = (
+            rte_components_in_exp + html_cleaner.get_rte_components(
+                html_string))
+
+    def get_image_rte(rte_comp):
```

Comment:
Don't declare functions in functions. Also every function needs a full docstring and unit tests.

Reply:
Done!